src/v2i-hub/PedestrianPlugin/src/FLIRWebSockAsyncClnSession.cpp

```diff
@@ -317,10 +318,10 @@ namespace PedestrianPlugin
         std::cout << beast::make_printable(buffer_.data()) << std::endl;
     }
 
-    //returns the variable containing the psm xml 
-    string FLIRWebSockAsyncClnSession::getPSMXML() const
+    std::queue<std::string>& FLIRWebSockAsyncClnSession::getPSMQueue()
     {
-        return psmxml;
+        std::lock_guard<mutex> lock(_psmLock);
```

Comment:
You don't need a lock here because you are only returning the reference to the queue.  Doesn't hurt but it's not needed.